flair/datasets/sequence_labeling.py

```diff
@@ -628,6 +661,9 @@ def _read_next_sentence(self, file):
             if len(lines) > 0 and self.__line_completes_sentence(line):
                 break
 
+            if line.strip() == self.document_separator_line:
+                break
```

Comment:
What does this do? Can you add a comment?

Reply:
I am assuming document boundaries are separate sentences. Since there is no newline after the document separator line in CoNLL2012, the current sentence (i.e. document boundary) is completed directly.
This code should be moved to ColumnDataset.__line_completes_sentence (I fixed this in one of the commits).